test/extended/util/framework.go

```diff
@@ -219,7 +221,20 @@ func WaitForADeployment(client kclient.ReplicationControllerInterface,
 		defer w.Stop()
 
 		for {
-			val, ok := <-w.ResultChan()
+			timeout := make(chan bool, 1)
+			go func() {
+				time.Sleep(10 * time.Second)
```

Comment:
@rhcarvalho the watch may never expire, so we need to periodically stop and see if we've waited too long.